### PR TITLE
api: add endpoint to handle k8s service account in cluster and download associated kubeconfigs

### PR DESCRIPTION
**Pull request by vgramer**
ThisPR is follow-up of #11096,  it adds API endpoints to create service accounts and get associated kubeconfig.

Service accounts are labeled with `component=clusterServiceAccount` to avoid deleting critical service accounts and show relevant ones in the dashboard.

When the service account is deleted, we remove all binding from clusterRoleBinding and RoleBinding labeled `component=userClusterBinding` (I.e. binding that have been created through kubermatic rbac api)

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref #10166

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
adds API endpoints to create service accounts and get associated kubeconfig. 
 - GET, POST /api/v2/projects/{project_id}/clusters/{cluster_id}/serviceaccount
 - DELETE /api/v2/projects/{project_id}/clusters/{cluster_id}/serviceaccount/{namespace}/{service_account_id}
 - GET /api/v2/projects/{project_id}/clusters/{cluster_id}/serviceaccount/{namespace}/{service_account_id}/kubeconfig
```

**Documentation**:

```documentation
https://github.com/kubermatic/docs/pull/1310
```

### CODE REVIEW

**Comment by pkprzekwas** [comment]
v3?
> **Does this PR introduce a user-facing change? Then add your Release Note here**:
> 
> ```
> adds API endpoints to create service accounts and get associated kubeconfig. 
>  - GET, POST /api/v3/projects/{project_id}/clusters/{cluster_id}/serviceaccount
> ```

**Comment by vgramer** [comment]
@pkprzekwas oups my bad, fixed!

File: pkg/handler/v2/cluster/serviceaccount.go [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+	// ServiceAccountComponentKey is the label key that tag k8s service account created though UI.
+	ServiceAccountComponentKey = "component"
+
+	// ServiceAccountComponentValue is the value associated to ServiceAccountComponentKey label.
+	ServiceAccountComponentValue = "clusterServiceAccount"
+)
+
+func ListClusterSAEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listClusterSAReq)
```
Comment (pkprzekwas) [inline pkg/handler/v2/cluster/serviceaccount.go]: nit: consider adding error check for casting types
```suggestion
		req, ok := request.(listClusterSAReq)
```

File: pkg/handler/v2/cluster/serviceaccount.go [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+		for i := range serviceAccountList.Items {
+			saList[i] = convertInternalClusterSAToExternal(&serviceAccountList.Items[i])
+		}
+		return saList, nil
+	}
+}
+
+func CreateClusterSAEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(CreateClusterSAReq)
```
Comment (pkprzekwas) [inline pkg/handler/v2/cluster/serviceaccount.go]: nit
```suggestion
		req, ok := request.(CreateClusterSAReq)
```

File: pkg/handler/v2/cluster/serviceaccount.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+		}
+
+		serviceAccount := &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      req.Body.Name,
+				Namespace: req.Body.Namespace,
+				Labels:    map[string]string{ServiceAccountComponentKey: ServiceAccountComponentValue},
+			},
+		}
+		if err := client.Create(ctx, serviceAccount); err != nil { // todo
```
Comment (pkprzekwas) [inline pkg/handler/v2/cluster/serviceaccount.go]: Redundant comment

File: pkg/handler/v2/cluster/serviceaccount.go [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		return convertInternalClusterSAToExternal(serviceAccount), nil
+	}
+}
+
+func DeleteClusterSAKubeconigEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(ClusterSAReq)
```
Comment (pkprzekwas) [inline pkg/handler/v2/cluster/serviceaccount.go]: nit
```suggestion
		req, ok := request.(ClusterSAReq)
```

File: pkg/handler/v2/cluster/serviceaccount.go [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+		return nil, err
+	}
+	req.ClusterID = clusterID
+
+	return req, nil
+}
+
+// Validate validates listClusterSAReq request.
+func (req listClusterSAReq) Validate() error {
+	if len(req.ProjectID) == 0 {
```
Comment (pkprzekwas) [inline pkg/handler/v2/cluster/serviceaccount.go:215]: Is there any reason why `ClusterID` is not checked same way as `ProjectID`?

File: pkg/handler/v2/cluster/serviceaccount.go [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+	req.ProjectReq = projectReq.(common.ProjectReq)
+
+	if err := json.NewDecoder(r.Body).Decode(&req.Body); err != nil {
+		return nil, err
+	}
+	return req, nil
+}
+
+// Validate validates CreateClusterSAReq request.
+func (req CreateClusterSAReq) Validate() error {
```
Comment (pkprzekwas) [inline pkg/handler/v2/cluster/serviceaccount.go:261]: Same as above. Why `ClusterID` is not validated?

File: pkg/handler/v2/cluster/serviceaccount.go [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+	if saID == "" {
+		return "", fmt.Errorf("'service_account_id' parameter is required but was not provided")
+	}
+	req.ServiceAccountID = saID
+
+	return req, nil
+}
+
+// Validate validates ClusterSAReq request.
+func (req ClusterSAReq) Validate() error {
```
Comment (pkprzekwas) [inline pkg/handler/v2/cluster/serviceaccount.go:329]: Same as above. Why ClusterID is not validated?

File: pkg/handler/v2/routes_v2.go [hunk 8]
```diff
@@ -7835,3 +7852,109 @@ func (r Routing) listOperatingSystemProfilesForCluster() http.Handler {
...
+//
+//	Consumes:
+//	- application/json
+//
+//	Produces:
+//	- application/json
+//
+//	Responses:
+//	  default: errorResponse
+//	  200: ClusterServiceAccount
```
Comment (pkprzekwas) [inline pkg/handler/v2/routes_v2.go]: nit: it's a good practice to return 201 status after creating resources. Remember to update response encoder as well.
```suggestion
//	  201: ClusterServiceAccount
```

Comment (vgramer) [reply pkg/handler/v2/routes_v2.go]: done :)

File: pkg/handler/v2/cluster/serviceaccount.go (hunk 4 above)
Comment (vgramer) [reply pkg/handler/v2/cluster/serviceaccount.go]: done :)

File: pkg/handler/v2/cluster/serviceaccount.go (hunk 1 above)
Comment (vgramer) [reply pkg/handler/v2/cluster/serviceaccount.go]: done :)

File: pkg/handler/v2/cluster/serviceaccount.go (hunk 2 above)
Comment (vgramer) [reply pkg/handler/v2/cluster/serviceaccount.go]: done :)

File: pkg/handler/v2/cluster/serviceaccount.go (hunk 3 above)
Comment (vgramer) [reply pkg/handler/v2/cluster/serviceaccount.go]: done :)

File: pkg/handler/v2/cluster/serviceaccount.go (hunk 5 above)
Comment (vgramer) [reply pkg/handler/v2/cluster/serviceaccount.go:215]: opposite to `ProjectID` validation is already done in [DecodeClusterID](https://github.com/kubermatic/kubermatic/blob/84b83afc7770e15f18a44f0cc9e57d2bb9250656/pkg/handler/v1/common/request.go#L127-L129) function.

**Pull request merged**